Feature/sw 79 finish testcase for lenscalibration

JFSolution opened the pull request:
added testcases for boarddetection, lenscalibration and marker detection
Jira Ticket https://mr-beam.atlassian.net/browse/SW-79

Commit:
added automated tests

Commit:
fixed lens_calibration testcase
removed rsc files from git => they are in https://github.com/mrbeam/test_rsc repository
https://mr-beam.atlassian.net/browse/SW-79

Commit:
revert changes to __init__, camera and lens file, but calibrationState causes errors
fixed error with SYMLINK_IMG_DIR path if test is run not on a raspberry pi
https://mr-beam.atlassian.net/browse/SW-79

Commit:
Merge remote-tracking branch 'origin/master' into f_test_camera

# Conflicts:
#	octoprint_camera/__init__.py
#	octoprint_camera/corners.py
#	octoprint_camera/lens.py

amrsoll commented on octoprint_camera/camera.py:
```diff
@@ -67,6 +67,8 @@ def get_latest_img(self):
             self._logger.warning("Camera not initalised, cannot return an image")
             return None
 
+    #
+
```
Please rm

JFSolution replied:
removed

amrsoll commented on octoprint_camera/static/js/calibration/corner_calibration.js:
```diff
@@ -233,8 +233,10 @@ $(function () {
+                    console.log('get image corner - corner');
                     self.camera.getImage(GET_IMG.last, GET_IMG.pic_corner);
                 } else {
+                    console.log('get image corner - plain')
```
Please rm logs if they are no longer needed

amrsoll commented on octoprint_camera/static/js/calibration/watterott/calibration_qa.js:
```diff
@@ -25,9 +27,40 @@ $(function () {
             }
         });
-        self.croppedImgWidth = 500;
+        self.croppedImgWidth = "500";
```
Why is it converted into a string?

JFSolution replied:
hmm I don't know, this was not intended, will change it back

amrsoll commented on octoprint_camera/static/js/camera.js:
```diff
@@ -419,6 +419,8 @@ $(function () {
                 .join(" ");
         });
         self.zObjectImgTransform = ko.computed(function () {
+            console.log('zobjectimg trans', self.imgHeightScale(), self.imgTranslate(), self.workingAreaWidthMM(), self.workingAreaHeightMM(), 'imgheightscale params(',self.cornerMargin() *
+                (1 - self.objectZ() / self.maxObjectHeight), ';', self.cornerMargin(), self.objectZ(), self.maxObjectHeight, ')', self.defaultMargin);
```
Same as before when it comes to logging - please rm

JFSolution replied:
done

amrsoll commented on octoprint_camera/static/js/camera.js:
```diff
@@ -136,7 +136,7 @@ $(function () {
                 });
             };
-
+            // self.availablePicTypes['corners'](true);
```
Please rm

amrsoll commented on octoprint_camera/static/js/calibration/watterott/calibration_qa.js:
```diff
@@ -25,9 +27,40 @@ $(function () {
+                self.imgTranslate() +
+                ")"
+            );
+        });
```
Please do not introduce duplication. If you need these functions and observables, then import them from other viewmodels. In this case, use the `cameraViewmodel` - `self.camera`.

amrsoll replied:
You can also create a new pointer if you want the qa screen viewmodel to have `self.some_attr` :

```js
self.some_attr = self.camera.some_attr
```

This way you keep all the `ko` bindings that are used elsewhere.

Example:

```js
self.zObjectImgTransform = self.camera.zObjectImgTransform
```

amrsoll commented on tests/README.md:
```diff
@@ -0,0 +1,16 @@
+##prepare the test
+clone repository `git@github.com:mrbeam/test_rsc.git`, this contains the files needed for the test
```
please let us know if we need to clone it somewhere special or make a symlink from this repo

amrsoll reviewed (changes requested):
Great :) Please be careful about duplicity though.

Commit:
requested changes from PR#5 https://github.com/mrbeam/OctoPrint-Camera/pull/5
removed more debug logging
changed path for rsc folder to match with test_rsc repository filetree

Commit:
added missing newline for gitignore and pytest.ini
removed some unwanted changes

amrsoll reviewed (approved):
Great!
